### PR TITLE
Integrate rsmu

### PR DESCRIPTION
**Pull request by talyacs**
Manual integration of [renesas linux ptp driver package](https://github.com/renesas/linux-ptp-driver-package)
which includes support for the pcm4l software

### CODE REVIEW

**Comment by shmuelhazan** [comment]
@talyacs Could you please link here the source of this code?

File: drivers/ptp/ptp_idt82p33.c [hunk 1]
```diff
@@ -6,13 +6,17 @@
 #define pr_fmt(fmt) "IDT_82p33xxx: " fmt
```
Comment (shmuelhazan) [inline drivers/ptp/ptp_idt82p33.c:6]: filemode changed to 100755.

**Comment by talyacs** [comment]
> @talyacs Could you please link here the source of this code?

Done.

File: drivers/ptp/ptp_clockmatrix.h [hunk 2]
```diff
@@ -54,21 +54,9 @@
...
 
-#define IDTCM_MAX_WRITE_COUNT		(512)
-
 #define PHASE_PULL_IN_MAX_PPB		(144000)
 #define PHASE_PULL_IN_MIN_THRESHOLD_NS	(2)
 
-/*
- * Return register address based on passed in firmware version
- */
-#define IDTCM_FW_REG(FW, VER, REG)	(((FW) < (VER)) ? (REG) : (REG##_##VER))
```
Comment (shmuelhazan) [inline drivers/ptp/ptp_clockmatrix.h:65]: could you please check if this change has been upstreamed?

Comment (talyacs) [reply drivers/ptp/ptp_clockmatrix.h:65]: I checked on the latest 6.4 version and it's not been upstreamed

File: drivers/ptp/ptp_idt82p33.c (hunk 1 above)
Comment (talyacs) [reply drivers/ptp/ptp_idt82p33.c:6]: Fixed.

**Review by shmuelhazan — APPROVED** [review]

**Pull request merged**